mongodbatlas_test.go

```diff
@@ -50,30 +63,34 @@ func TestAcceptanceDatabaseUser_CreateUser(t *testing.T) {
 	}
 
 	db := new()
-	_, err := db.Init(context.Background(), connectionDetails, true)
-	if err != nil {
-		t.Fatalf("err: %s", err)
-	}
+	defer dbtesting.AssertClose(t, db)
 
-	statements := dbplugin.Statements{
-		Creation: []string{testMongoDBAtlasRole},
+	initReq := newdbplugin.InitializeRequest{
+		Config: connectionDetails,
 	}
 
-	usernameConfig := dbplugin.UsernameConfig{
-		DisplayName: "test",
-		RoleName:    "test",
+	dbtesting.AssertInitialize(t, db, initReq)
+
+	createReq := newdbplugin.NewUserRequest{
+		UsernameConfig: newdbplugin.UsernameMetadata{
+			DisplayName: "test",
+			RoleName:    "test",
+		},
+		Statements: newdbplugin.Statements{
+			Commands: []string{testMongoDBAtlasRole},
+		},
 	}
 
-	username, _, err := db.CreateUser(context.Background(), statements, usernameConfig, time.Now().Add(time.Minute))
+	createResp, err := db.NewUser(context.Background(), createReq)
 	if err != nil {
 		t.Fatalf("err: %s", err)
 	}
 
-	if err := testCredsExists(projectID, publicKey, privateKey, username); err != nil {
+	if err := testCredsExists(projectID, publicKey, privateKey, createResp.Username); err != nil {
```

Comment:
Github won't let me comment on where `testCredsExists` is being defined, so I'm putting this comment here:

`testCredsExists` doesn't appear to be checking to make sure the password for the given user is correct, it just checks if the user exists. This is particularly problematic in the `UpdateUser` case where the password is being changed.

Reply:
This is a bit tricker that I originally thought it would be since the response never returns the password and we're not currently creating an actual database/cluster for testing (DB users don't require one since they are project-based). I'll have to create and teardown a test cluster so that I can test username + password via an actual connection.